bin/multi2single_sequence.py

```diff
@@ -0,0 +1,58 @@
+#!/usr/bin/env python
+## Originally written by Andries van Tonder and released under the MIT license.
+## See git repository (https://github.com/nf-core/bactmap) for full license text.
+
+from Bio import SeqIO
+from Bio.Seq import Seq
+from Bio.SeqRecord import SeqRecord
+import textwrap
+import argparse, sys, os
+
+class ParserWithErrors(argparse.ArgumentParser):
+    def error(self, message):
+        print('{0}\n\n'.format(message))
+        self.print_help()
+        sys.exit(2)
+
+    def is_valid_file(self, parser, arg):
+        if not os.path.isfile(arg):
+            parser.error("The file %s does not exist!" % arg)
+        else:
+            return arg
+
+
+def argparser():
+    description = """
+    A script to parse a multiple contig fasta and
+    concatenate the sequences into a single sequence.
+    The new sequence will be output to a new fasta file.
+    """
+    parser = ParserWithErrors(description = description)
+    parser.add_argument("-r", "--reference_file", required=True,
+                        help="reference fasta file path",
+                        type=lambda x: parser.is_valid_file(parser, x))
+    parser.add_argument("-o", "--output_fasta_file", required=True,
+                    help="file path to output fasta file")
+
+    return parser
+
+def combine_sequences(reference_sequence):
+    records = list(SeqIO.parse(reference_sequence, "fasta"))
+    new_sequence = ''.join([str(record.seq) for record in records])
+    new_id = '|'.join([record.id for record in records])
```

Comment:
nitpicking
If the Multifasta is really, really big, I thought that one might encounter OOM by converting it to a list which reads it all into mem. If you are confident that this will never occur, ignore this. If so maybe something like this? Haven't tested this
```suggestion
    with open("merged.fasta", "w") as f:
    f.write(">")  # header placeholder

    for record in SeqIO.parse(reference_sequence, "fasta"):
        ids.append(record.id)
        f.write(str(record.seq))

    # rewrite header with all IDs at the start
    new_id = '|'.join(ids)
    if len(new_id) > 100:
        new_id = new_id[:97] + '...'
    f.seek(0)
    f.write(new_id + "\n")
 ```

Reply:
The pipeline is designed for mapping sequence data to bacterial reference sequences. The genomes are never going to be more than 10Mb.